File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];
        int num = 0;
        
        for (int[] que : queries) {
            int tmp = 0;
                
            for(int i = que[0]; i <= que[1]; i++) {
                if (que[2] < arr[i]) {
                    if (tmp == 0 || arr[i] < tmp) tmp = arr[i];
                }
            }
            System.out.println(tmp);
            if (tmp == 0) tmp = -1;
                
            answer[num] = tmp;
            num += 1;
        }
        return answer;
    }
}

File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []
    
    for s, e, k in queries:
        print(s, e, k)
        tmp = 0
        for i in range(s, e+1):
            if k < arr[i]:
                if tmp == 0 or arr[i] < tmp: tmp = arr[i]
            
        if tmp == 0: answer.append(-1)
        else: answer.append(tmp)
                
    return answer